src/codegen/generators/data_type_generator.rs

```diff
@@ -131,6 +131,18 @@ fn create_type<'ink>(
             let gen_type = llvm.context.i8_type().array_type(*size).into();
             Ok(gen_type)
         }
+        DataTypeInformation::SubRange {
+            referenced_type, ..
+        } => {
+            let ref_type = create_type(
+                llvm,
+                index,
+                types_index,
+                name,
+                index.find_type(referenced_type).unwrap(),
```

Comment:
Is that enough? It won't follow aliases, but I'm assuming in this situation we don't need to because subranges are only valid for Primitive types?